elastica/experimental/connection_contact_joint/generic_system_type_connection.py

```diff
@@ -0,0 +1,414 @@
+            point_system_one
+            if point_system_one is not None
+            else np.array([0.0, 0.0, 0.0])
+        )
+        self.point_system_two = (
+            point_system_two
+            if point_system_two is not None
+            else np.array([0.0, 0.0, 0.0])
+        )
+
+    def apply_forces(self, system_one, index_one, system_two, index_two):
+        """
+        Apply joint force to the connected systems.
+
+        Parameters
+        ----------
+        system_one : SystemType
+             System two of the joint connection.
+             Object of a child class of  either `CosseratRod` or `RigidBodyBase`.
+        index_one : int
+            Index of first system for joint.
+        system_two : SystemType
+            System two of the joint connection.
+            Object of a child class of  either `CosseratRod` or `RigidBodyBase`.
+        index_two : int
+            Index of second system for joint.
+
+        Returns
+        -------
+
+        """
+        # Compute the position in the inertial frame of the specified point.
+        # The point is defined in the local coordinate system of system one and used to attach to the joint.
+        position_system_one = compute_position_of_point(
+            system=system_one, point=self.point_system_one, index=index_one
+        )
+        # Compute the position in the inertial frame of the specified point.
+        # The point is defined in the local coordinate system of system two and used to attach to the joint.
+        position_system_two = compute_position_of_point(
+            system=system_two, point=self.point_system_two, index=index_two
+        )
+
+        # Analogue to the positions, compute the velocities of the points in the inertial frames
+        velocity_system_one = compute_velocity_of_point(
+            system=system_one, point=self.point_system_one, index=index_one
+        )
+        velocity_system_two = compute_velocity_of_point(
+            system=system_two, point=self.point_system_two, index=index_two
+        )
+
+        # Compute the translational deviation of the point belonging to system one
+        # from the point belonging to system two
+        end_distance_vector = position_system_two - position_system_one
+
+        # Calculate norm of end_distance_vector
+        # this implementation timed: 2.48 µs ± 126 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
+        end_distance = np.sqrt(np.dot(end_distance_vector, end_distance_vector))
+
+        # Below if check is not efficient find something else
+        # We are checking if the point of `system_one` and point of `system_two` are at the same point in space
+        # If they are at the same point in space, it is a zero vector.
+        if end_distance <= Tolerance.atol():
+            normalized_end_distance_vector = np.array([0.0, 0.0, 0.0])
+        else:
+            normalized_end_distance_vector = end_distance_vector / end_distance
+
```

Comment:
```suggestion
```

Reply:
Implemented in https://github.com/GazzolaLab/PyElastica/pull/149/commits/cc2010ddebfff9cf7128930dfb6e6e9ad92eb167